### PR TITLE
[Tests] [JSight to OpenAPI converter] req.japi.http_method.query_0.3

### PR DESCRIPTION
**Pull request by bereg2k**
* New Functional Tests for OpenAPI Converter: "req.japi.http_method.query_0.3"

### CODE REVIEW

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          {
+            "in": "query",
+            "name": "prim2",
+            "style": "deepObject",
+            "explode": true,
+            "schema": {
+              "type": "string",
+              "example": "value"
+            },
+            "required": false
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:31]: We do not show properties by default, let's get rid of this here and in other places within the pull request.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          {
+            "in": "query",
+            "name": "arr",
+            "style": "deepObject",
+            "explode": true,
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "integer"
+              }
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:42]: Array is not properly converted. You cannot lose the example values (see the convert rules docs https://j-sight.atlassian.net/wiki/spaces/JAPI/pages/402915358#array). It should be:

```
  "items": {
    "anyOf": [
      {
        "type": "integer",
        "example": 2
      },
      {
        "type": "integer",
        "example": 3
      }
    ]
  }
```

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          },
+          "prim2": {
+            "type": "string",
+            "example": "value"
+          },
+          "arr": {
+            "type": "array",
+            "items": {
+              "type": "integer"
+            }
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo:42]: The array is incorrectly converted. `anyOf` must be used, see https://j-sight.atlassian.net/wiki/spaces/JAPI/pages/402915358#array.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+          {
+            "in": "query",
+            "name": "second",
+            "style": "deepObject",
+            "explode": true,
+            "schema": {
+              "type": "string",
+              "example": "value"
+            },
+            "required": false
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo:102]: We do not show default OpenAPI values.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+          {
+            "in": "query",
+            "name": "arr",
+            "style": "deepObject",
+            "explode": true,
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "integer"
+              }
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo:151]: The array is incorrect, see the same issues above.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type_allOf.openapi.todo [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+          },
+          {
+            "$ref": "#/components/schemas/queryUserType2"
+          }
+        ]
+      }
+    }
+  },
+  "paths": {
+    "/query/one-user-type": {
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type_allOf.openapi.todo:69]: In the source jsight document, there is no `/query/one-user-type` path, but two `/query/one-user-type-1` and `/query/one-user-type-2` paths.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            },
+            "required": false
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:44]: Get rid of it.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                  "type": "integer",
+                  "example": 45
+                }
+              },
+              "required": [
+                "foo"
+              ],
+              "additionalProperties": false
+            },
+            "required": false
```
Comment (andrew-platonov) [inline testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:73]: Get rid if it.

**Review by andrew-platonov — CHANGES_REQUESTED** [review]

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo (hunk 1 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:31]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo (hunk 8 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:73]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo (hunk 3 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo:42]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo (hunk 4 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type.openapi.todo:102]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo (hunk 2 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:42]: Fixed.

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type_allOf.openapi.todo (hunk 6 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_user_type_allOf.openapi.todo:69]: Fixed in `query_user_type_allOf.jst` by leaving only one  `GET /query/one-user-type`

File: testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo (hunk 7 above)
Comment (bereg2k) [reply testdata/jsight_0.3/ser-to-openapi/req.japi.http_method.query_0.3/query_basic.openapi.todo:44]: Fixed.

**Review by andrew-platonov — APPROVED** [review]

**Pull request merged**